day34/validateIP.py

```diff
@@ -0,0 +1,74 @@
+def validateIP(ip):
+    ipv4 = ip.split(".")
+    ipv6 = ip.split(":")
+    if len(ipv4) == 4 and validateIPv4(ipv4):
+        return "IPv4"
+    elif len(ipv6) == 8 and validateIPv6(ipv6):
+        return "IPv6"
+    else:
+        return "Neither"
+
+def validateIPv4(segments):
+    digits = set("1234567890")
+    for s in segments:
+        # segment too long or too short
+        if len(s) > 3 or len(s) == 0:
+            return False
+
+        # segment has leading 0
+        if len(s) > 1 and s[0] == "0":
+            return False
+
+        # segment contains non-numeric digits
+        if not set(s) < digits:
+            return False
+
+        # segment is not an int between 0 and 255 inclusive
+        if int(s) > 255 or int(s) < 0:
+            return False
+    return True
+
+def validateIPv6(segments):
+    hexDigits = set("1234567890abcdefABCDEF")
+    for s in segments:
+        # segment too long or too short
+        if len(s) > 4 or len(s) == 0:
+            return False
+
+        # segment contains non-hexadecimal digits
+        if not set(s) < hexDigits:
```

Comment:
i'm starting to think it's better to do `cond1 = len_check; cond2 = characters_check; return cond1 and cond2 and cond3...` instead of breaking up the flow of the program so much with so many `return False` lines